sql/plan/subquery.go

```diff
@@ -46,12 +46,12 @@ type Subquery struct {
 	// reflect.DeepEquals, but it's safe in practice because the function is always nil until execution.
 	disposeFunc sql.DisposeFunc
 	// Mutex to guard the caches
-	cacheMu sync.Mutex
+	cacheMu *sync.Mutex
```

Comment:
Do we want this? Usually it doesn't make sense to e.g. share a mutex when you deref a pointer to a struct, which will happen if we do this

Reply:
TBH I wasn't sure, but I thought since the `hashCache` is a pointer that all access to it should probably use the same mutex, so we should carry the mutex as well?

Reply:
Technically the hash cache has the same problem: we need to clone it when returning a copy of this node to be safe. But let's not make the problem any worse.

(This doesn't matter in practice because we don't hang onto multiple copies of these nodes, it's just best practice)

Reply:
Good point!